RecoJets/JetProducers/plugins/VirtualJetProducer.cc

```diff
@@ -456,9 +470,29 @@ void VirtualJetProducer::inputTowers() {
 	std::cout << "PF cand:" << pfc << '\n';
       }
       */
-      fjInputs_.emplace_back(input.px(), input.py(), input.pz(), input.energy());
+      if (!applyWeight_) {
+        fjInputs_.emplace_back(input.px(), input.py(), input.pz(), input.energy());
+        fjInputs_.back().set_user_index(i - inBegin);
+      } else {
+        float w = 0.0;
+        if (!input_weights_token_.isUninitialized())
+          w = weights_[*i];
+        else {
+          pat::PackedCandidate const* pPC = dynamic_cast<pat::PackedCandidate const*>(i->get());
+          if (pPC) {
+            w = pPC->puppiWeight();
+            weights_[*i] = w;
```

Comment:
similar to other cases, I think that it's better to always use an external weight